### DISCUSSION TITLE
Add ability to create multiple sensors and have a unique id

### DISCUSSION THREAD

**Discussion by TheRealWaldo** [TheRealWaldo/thermal#178] [Ideas]
Currently does not have a unique ID so cannot customize entities via. the UI.

Allow configuration with bulk syntax to allow for unique IDs to be set easier.

**Comment by emdekext** [comment]
Great! Just finished installing my first thermal cam custom integration today, the Eyalcha version. It works with my AMG8833 prototype, but not without several modifications. Started a couple of days ago, so did not notice your fork until today, probably would have saved me a lot of work (but I learned some valuable lessons though). I will switch over to your fork next weekend, I'll keep you posted. 
Just received an Omron D6T44L06, will try to set this up next weekend also.

**Comment by TheRealWaldo** [comment]
@emdekext awesome!  Let me know how that goes.

I've been looking at some of the higher resolution sensors as well, not sure how well this integration work with them yet, but please share your source if you get it working!

I'll also be open sourcing my version of the ESP firmware soon.

**Comment by emdekext** [comment]
The camera works in Home Assistant with both yours and Eyalcha's custom components simultaneously, but I can't get the sensor to work with your integration, no readings at all (works with Eyalcha's, but with the know issue of mixed up min and max values). Before I do any further research: I'm using a modified version of Eyalcha's grideye.ino webserver (modified it for use with an esp32), can you give me any info on your ESP firmware? Thanks!

**Comment by TheRealWaldo** [comment]
@emdekext I will do my best to get my version pushed to GitHub today; any errors/warnings in the HA log?

**Comment by TheRealWaldo** [comment]
@emdekext see https://github.com/TheRealWaldo/esp8266-amg8833

Very much a work-in-progress!

**Comment by TheRealWaldo** [comment]
@emdekext I have a theory about why it's not working for you, and I am working on proving it and a fix as we speak.

**Comment by emdekext** [comment]
Thanks! I don't have much spare time right now but I did not notice any errors/warnings in the HA log, I checked. Debug logging has to wait a couple of days. If I can provide some quick additional info let me know, I'll try to answer asap.

**Comment by TheRealWaldo** [comment]
@emdekext 2 things I noticed when I reverted the firmware to Eyalcha's:

* I added the sensor temperature to my sensor firmware and attempted to make the integration backward-compatible but did not properly check.
* When the sensor temperature didn't exist, and the client was set to °F, the sensor failed to update.

I've just pushed fixes for both. If tests pass, I'll have a new version released in a few minutes.

**Comment by TheRealWaldo** [comment]
@emdekext should be fixed in v2.1.1!

**Comment by emdekext** [comment]
Well since released a fix so quickly I couldn't let that pass, so did a quick and dirty file replacement and restart of HA, and **it is fixed indeed!** I do have temp readings now. Great job, end of the week I'll do some more testing and take a look at the firmware you released. My first goal is to make a presence sensor to control lights (and other devices or automations), it annoys me that when using a PIR sensor there always has to be a timeout to prevent lights going out when not moving for a while, but when leaving the room the lights stay on for a couple of minutes without any purpose.
Did not get to testing the D6T sensor yet.

**Comment by TheRealWaldo** [comment]
I have similar plans and will be updating the firmware and integration to do a better job of presence detection.  I don't like that it's a polling model, so I intend to have the detection closer to the firmware and push updates to HA instead!

I've been collecting data on a few sensors in my home to figure out the best way to do this!  Please share your findings as well, because it doesn't look to be as simple as setting a 'high.'

**Comment by johanvh83** [comment]
Hi guys,
Thanks for your effort @TheRealWaldo. I'm not a programmer and ordered the AMG8833, and thanks to your well explained information I have everything up and running but the HA person presence entity is not reliable. Even when nobody is inside my house, the sensor gives 'occupied'. Found this thread so I was wondering if something changed last months

**Comment by TheRealWaldo** [comment]
@johanvh83 everything is in the sensor, but it is still very rudimentary.

Right now it takes a look at the difference between high and low temperatures.

So, depending on where your sensor is aimed, you may have something warmer than average in the view of the sensor.

Play with the values in the sensor code, and the positioning of your sensor.  I believe the default difference is 3 degrees.
